students/jared_mulholland/lesson_4/mail_room.py

```diff
@@ -0,0 +1,130 @@
+"""
+import numpy as np
+
+def takeSecond(elem):
+    return elem[1]
+
+def create_report():
+    rows = []
+    for donor in donors:
+        rows.append([donor, sum(donors[donor]), len(donors[donor]), np.mean(donors[donor])])
+
+    rows.sort(key = takeSecond, reverse = True)
+
+    print('{:<20s} |{:>15s}|{:>12s} |{:>15s}'.format('Donor Name','Total Given','Num Gifts','Average Gift'))
+    for row in rows:
+        print('{:<20s} ${:15,.2f} {:12d} ${:15,.2f}'.format(*row))
+
+
+"""
+Send Letters Function
+
+In this version, add a function (and a menu item to invoke it), that goes through all 
+the donors in your donor data structure, generates a thank you letter, and writes it to disk as a text file.
+"""
+import os 
+
+file_path = 'C:\\Users\\Jared\\Documents\\IntroToPython\\Self_Paced-Online\\students\\jared_mulholland\\lesson_4\\donation_letters'
+
+def send_letters():
+    file_path = input("\nPlease Enter File Path: ")
+    os.chdir(file_path)
+
+    for donor in donors:
+        
+        letter_text = f"""\nDear {donor.split()[0]}, \n\nThank you for your generous donation of ${sum(donors[donor]):,.2f}. Thanks to you we will finally be able to begin construction 
+    on the {donor.split()[1]} Memorial Children's wing at Dark Place Hospital\n\nYours Truly,\nJared Mulholland"""
+
+        with open(donor.replace(" ","_").lower() + '_donations.txt', 'w') as donation_letter:
+            donation_letter.write(letter_text)           
+
+
+"""
+PROGRAM OVERVIEW
+Write a small command-line script called mailroom.py. This script should be executable. The script should accomplish the following goals:\
+-It should have a data structure that holds a list of your donors and a history of the amounts they have donated. 
+    This structure should be populated at first with at least five donors, with between 1 and 3 donations each.
+-You can store that data structure in the global namespace.
+-The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+"""
+
+#donors = ['Bill Gates','Mark Zuckerberg','Jeff Bezos','Paul Allen','Mookie Betts']
+#donations = [[10000,12000],[20000,150000],[800],[100,500,200],[1000000,1500000,1750000]]
+
+donors = {'Bill Gates': [10000.0,12000.0],
+                'Mark Zuckerberg':[80000.0],
+                'Jeff Bezos':[30000.0,75000.0,35000.0],
+                'Paul Allen': [100000.0,130000.0],
+                'Mookie Betts': [150000.0,124000.0]}
+
+main_dict = {
+            "1": send_thankyou,
+            "2": create_report,
+            "3": send_letters
+            }
+    
+main_prompt = ("\nMain Menu  \n 1. Send a Thank You \n 2. Create a Report \n 3. Send Letters \n 4. Quit \n Please Choose an Option: ")
+
+def mail_room_fun(main_prompt, main_dict):
+    while True:        
+        response = input(main_prompt)
+        if response == "4":
+            break
+        else:
+            print("\n")
```

Comment:
in order to avoid this special logic you can map option "4" to a `quit_program` function that calls `sys.exit`